content/testing.md

```diff
@@ -0,0 +1,224 @@
+
+If you are testing that multiple modules behave properly in concert, you are writing an *integration test*. Such tests are much more complex and may require running code both on the client and on the server to verify that communication across that divide is working as expected. Typically an integration test will still isolate a part of the entire application and directly verify results in code.
+
+If you want to write a test that can be run against any running version of your app and verifies at the browser level that the right things happen when you push the right buttons, then you are writing an *acceptance* or *end-to-end (e2e) test*. Such tests typically try to hook into the application as little as possible, beyond perhaps setting up the right data to run a test against.
+
+Finally you may wish to test that your application works under typical load or see how much load it can handle before it falls over. This is called a *load test* or *stress test*. Such tests can be challenging to set up and typically aren't run often but are very important for confidence before a big production launch.
+
+<h3 id="isolation-techniques">Isolation techniques</h3>
+
+<h3 id="challenges-with-meteor">Challenges of testing in Meteor</h3>
+
+
+<h2 id="generating-test-data">Generating test data</h2>
+
+
+## Unit testing in Meteor
+
+Unit testing is the process of isolating a module of code and then testing that the internals of that module work as you expect.
+
+As we've organized our application into modules based on features, collections and utilities, it is natural to use that same breakdown to test those modules.
+
+### Unit testing with Mocha
+
+We'll use the popular [Mocha](https://mochajs.org) test runner alongside the [Chai](http://chaijs.com) assertion library to test our application. In order to write tests in Mocha, we can add the [`avital:mocha`](https://atmospherejs.com/avital/mocha) package to our app.
+
+```bash
+meteor add avital:mocha
+```
+
+This package provides a way for test files to register tests and for a [test driver](#test-driver) to run them.
+
+### Defining a Mocha test
+
+In the Todos example app, we have a special `TodosCollection` that sets a `createdAt` field whenever we insert a new todo item. 
+
+```js
+class TodosCollection extends Mongo.Collection {
+  insert(doc, callback) {
+    doc.createdAt = doc.createdAt || new Date();
+    return super(doc, callback);
+  }
+}
+
+export default Todos = new TodosCollection('Todos');
+```
+[`imports/todos/Todos.js`]
+
+We should test that the `Todos` collection indeed behaves as we expect and sets that field when we insert a doc. To do that, we can write a file `imports/todos/todos.tests.js`, and define a Mocha test in it:
+
+```js
+import { mocha } from "avital:mocha";
```

Comment:
This is wrong, `avital:mocha` does not import mocha. See: https://github.com/avital/meteor-mocha-for-1.3/issues/1